commit ee1212ce5217b6792f5e1ed78adfa3607105155e
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2019-06-14T13:59:22+02:00 (4 months ago)
    
    TRITON-1736 Add relevant error message when creating more than 8 disks in Flexible Disk instance

diff --git a/lib/do_instance/do_disk/do_add.js b/lib/do_instance/do_disk/do_add.js
index bde9438..1a645fa 100644
--- a/lib/do_instance/do_disk/do_add.js
+++ b/lib/do_instance/do_disk/do_add.js
@@ -58,6 +58,12 @@ function do_add(subcmd, opts, args, cb) {
                     return;
                 }
 
+                if (disks.length === 8) {
+                    next(new errors.UsageError(
+                        'Cannot add more than 8 disk to a VM'));
+                    return;
+                }
+
                 ctx.disks = disks;
                 next();
             });
@@ -104,9 +110,9 @@ function do_add(subcmd, opts, args, cb) {
                     console.log('Added disk "%s" in %s', disk.id,
                                 common.humanDurationFromMs(duration));
                     next();
-                } else {
-                    next(new Error('Failed to create disk'));
+                    return;
                 }
+                next(new Error('Failed to create disk'));
             });
         }
     ]}, cb);
